### DISCUSSION TITLE
storyboard tahoma2d

### DISCUSSION THREAD

**Discussion by Dixxi0509** [tahoma2d/tahoma2d#1314] [Ideas]
small idea 
i think it would be a good idea to add a storyboard layout for tahoma2d not sure if it would be a spereate room or it can be a tool in the tool bar

**Comment by llloveletter** [comment]
What I can gather from researching this topic, is that the level strip seems to be the prime candidate for implementing storyboarding features. Editable text that can be attributed to a frame is also a missing element. Something else to consider is how OT/T2D handles large scene files (my experience is mixed). 

There is one storyboarding tool that I feel gets everything right, and that's Griffith. I'm using the ideas as presented in this video to workshop features for T2D: https://www.youtube.com/watch?v=RWdw9rDRf7A. Hopefully, you can also see the same promise as I do in this tool's design.

Working with the idea that we will have a 'master scene' for storyboarding, this would be my 'wish list' for storyboarding development:

---

- "In-canvas frames" as presented in the video above **[(7:39)](https://youtu.be/RWdw9rDRf7A?t=459)**
  - Like onion skinning, but formats surrounding unique frames like an e-conte sheet centered on your camera
  - COULD be a short term replacement for the level strips' proposed usage for storyboarding (but will still hold its' own).
  - NOTE the immediate ability to compare frame compositions, and a focus on the camera rather than the whole canvas.
- Add a way to generate scenes based on special flags attributed to a row on the xsheet
  - Maybe some sort of scene start/stop marker that allows for scene IDs, and position swapping
    - (OR some kind developer already knows how to implement this efficiently/effectively/simply)
  - Will have to address differences in the storyboard and target scenes, like the resolution.
  - LIKELY lots of conflict with already populated projects, BEWARE lost drawings during testing
- "Overview interface" _similar_ to how it's presented in the video above **[(8:40)](https://youtu.be/RWdw9rDRf7A?t=520)**
  - Imagine the Preproduction Board panel, but it displays only the unique, rendered frames.
  - COULD be a dedicated storyboard panel, allowing for both time management, and order management.
  - In the short-term, can likely be implemented as a specialized export or render option.
- Add ability to dynamically change the timing of unique frames within a scene (Krita's storyboarding feature, for example).
  - We need a timing tool that's a step zoomed out from the Xsheet. 
    - Maybe condense the Xsheet into it's most basic components, and make a new panel from that. Either something like this: ![temp-screenshot](https://github.com/user-attachments/assets/5fda07f1-9daf-43bd-bd94-9ff49e1f1261) , or something like the level strip, with the editable information integrated into it, like Griffith's design: ![temp-screenshot](https://github.com/user-attachments/assets/d941c3df-e0e3-484c-b14a-dee1c99dd5ab)
  - THINK about how this feature accounts for separate levels/columns. Krita's storyboarding docker is pretty bad at this.

---

Share your thoughts, cause I really want storyboarding tools.

EDIT (2024-08-08): Thinking on it from this perspective, storyboarding features might be easier to implement as a new level, rather than a new scene type? Not sure.

EDIT (2024-09-25): Been using the Pre-Production board for a project I've been working on, and found it's actually very convenient for Layout work. Did some digging, and the [initial PR](https://github.com/opentoonz/opentoonz/pull/3665) actually outlines everything needed to make it useful for storyboarding. I'll leave my mess of thoughts here as-is, but I think the initial PR is the most actionable path. I would still like to see design elements from Griffith be considered early into development though, as it does have input from professional animators to guide it.
